flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/HiveParser.java

```diff
@@ -207,6 +232,23 @@ public List<Operation> parse(String statement) {
         }
     }
 
+    private Operation processSetCmd(HiveConf hiveConf, Matcher setCmdMatcher) {
+        if (setCmdMatcher.group("verbose") != null) {
+            throw new UnsupportedOperationException("Unsupported command 'set -v'.");
+        }
+        String key = setCmdMatcher.group("key");
+        String value = setCmdMatcher.group("val");
+        if (FLINK_RESERVED_KEYWORDS_FOR_SET_CMD.contains(key)) {
+            // try to let Flink CLI handle it
+            return new SetOperation(key, value);
+        } else {
+            if (!key.equals("silent")) {
+                HiveParserSetProcessor.setVariable(hiveConf, hiveVariables, key, value);
+            }
+            return new NopOperation();
```

Comment:
We'd better to also set the config into table config because users may set some optimization options here.